test_socket.py

```diff
@@ -0,0 +1,140 @@
+import pytest
+import socket
+import string
+import email.utils
+
+
+@pytest.fixture(scope='session')
+def connection():
+    ADDR = ('127.0.0.1', 9992)
+    client = socket.socket(
+        socket.AF_INET, socket.SOCK_STREAM, socket.IPPROTO_IP
+    )
+    client.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+    client.connect(ADDR)
+    return client
+
+
+def test_wrongbreak():
+    ADDR = ('127.0.0.1', 9992)
+    client = socket.socket(
+        socket.AF_INET, socket.SOCK_STREAM, socket.IPPROTO_IP
+    )
+    client.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+    client.connect(ADDR)
+    message = "SET /index.html HTTP/1.1" + "\n" + "Host: www.example.com"
+    client.sendall(message)
+    client.shutdown(socket.SHUT_WR)
+    part = ""
+    while True:
+        tempPart = client.recv(16)
+        part = part + tempPart
+        if len(tempPart) < 16:
+            client.close()
+            break
+    part_list = string.split(part, '\r\n')
+    assert part_list[0].find("400 Bad Request") != -1
+
+
+def test_all(connection):
+    message = "GET /index.html HTTP/1.1" + "\r\n" + "Host: www.example.com"
+    connection.sendall(message)
+    connection.shutdown(socket.SHUT_WR)
+    part = ""
+    while True:
+        tempPart = connection.recv(16)
+        part = part + tempPart
+        if len(tempPart) < 16:
+            connection.close()
+            break
+    part_list = string.split(part, '\r\n')
+    assert part_list[0].find("200 OK") != -1
+    date = email.utils.formatdate(usegmt=True)
```

Comment:
This is fragile.  What happens if a second ticks over between when the server builds its response and when your test evaluates the response.  Don't match exact times.  It never ends well in tests.